racetrack_commons/racetrack_commons/plugin/engine.py

```diff
@@ -129,14 +130,29 @@ def on_moved(self, event):
         observer.schedule(event_handler, path=Path(self.plugins_dir).as_posix(), recursive=False)
         observer.start()
 
-    def upload_plugin(self, filename: str, file_bytes: bytes):
+        # inotify may not work on network filesystems
+        def _check_periodically():
+            while True:
+                time.sleep(5 * 60)
+                try:
+                    compare_changes()
+                except BaseException as e:
+                    log_exception(ContextError('periodic check failure', e))
+
+        threading.Thread(target=_check_periodically, daemon=True).start()
+
+
+    def upload_plugin(self, filename: str, file_bytes: bytes) -> PluginManifest:
         target_zip = Path(self.plugins_dir) / filename
         assert target_zip.suffix == '.zip', '.zip plugins are only supported'
 
         # save to tmp.zip to avoid overwriting current plugins
         tmp_zip = Path(self.plugins_dir) / 'tmp.zip'
+        if not tmp_zip.is_file():
+            tmp_zip.touch()
+            tmp_zip.chmod(mode=0o666)
         tmp_zip.write_bytes(file_bytes)
```

Comment:
I guess that when tmp.zip already exists, then write_bytes will overwrite it with new content?

Reply:
Do you think it's something be should bother?

Reply:
It's not something likely to happen, but deleting the file upfront, plus generating few chars of random name ie. tmp_123456.zip makes this situation impossible. One less problem to worry about or debug afterwards :) and it's 3 lines of code.

Reply:
Right, random name is a good way to go.